src/sdk/namespace_fs.js

```diff
@@ -1427,15 +1427,16 @@ class NamespaceFS {
                 dbg.log1('Namespace_fs._move_to_dest_version:', latest_ver_info, new_ver_info, gpfs_options);
 
                 if (this._is_versioning_suspended()) {
-                    if (latest_ver_info?.version_id_str === NULL_VERSION_ID) {
+                    //on GPFS safe_move overrides the latest object so no need to unlink
+                    if (!is_gpfs && latest_ver_info?.version_id_str === NULL_VERSION_ID) {
```

Comment:
I think that this condition is not good because if it's GPFS but it's also NULL_VERSION_ID, we will do the else, I think that on the GPFS machine you didn't have this issue

Reply:
right I changed it to be a condition inside the is latest version null condition. this way if the version isn't null we will act the same, and if it is null we will no unlink only on GPFS machine